feat: adds discord notification

galanggg opened the pull request:
Hello everyone, I made this pr so that Monika can send notifications via Discord. Please kindly check and review it thanks.

kevinhermawan commented:
Hello @galanggg, thank your for the PR we're really appreciate your hard work on this section 🍻

Have you update the documentation and add `discord` in the [notification section](https://github.com/hyperjumptech/monika/blob/main/docs/src/pages/guides/notifications.md)?

dennypradipta commented on src/components/notification/checker.ts:
```diff
@@ -188,6 +204,11 @@ export const notificationChecker = async (notifications: Notification[]) => {
     .map(webhookNotificationInitialChecker)
 
+  const discordWebhookNotification = notifications
+    .filter((notif) => notif.type === 'discordWebhook')
```
Hi @galanggg thank you for your contribution!

Is there any other way to connect into Discord besides Webhook? If there is not, I think it's better renamed as `discord` rather than `discordWebhook` to make it uniform like other types (e.g `slack`, `teams`, `discord`)

galanggg replied:
Hi hi @dennypradipta, thank you for the review!

For now, Discord only supports webhook. Yup, I also think it's better to change to `discord` to make it look clearer. Thank you for the feedback also :pray:

galanggg commented:
> Hello @galanggg, thank your for the PR we're really appreciate your hard work on this section 
> 
> Have you update the documentation and add `discord` in the [notification section](https://github.com/hyperjumptech/monika/blob/main/docs/src/pages/guides/notifications.md)?

Hello @kevinhermawan, thank you so much I really appreciate it. :pray: 

Sure I will update the documentation for discord notification.

galanggg commented:
Hi @kevinhermawan @dennypradipta, I just added the documentation for discord and changed the name from `discordWebhook` to `discord` also add some tests for discord notification.

kevinhermawan reviewed (approved):
Looks good to me!

dennypradipta reviewed (approved):
Well done!

dennypradipta reviewed (approved):
Well done!

galanggg commented:
Thanks all :rocket:
